src/hazelcore/dynamics/Evaluator.re

```diff
@@ -816,4 +829,31 @@ and evaluate_case =
         evaluate_case(inconsistent_info, scrut, rules, current_rule_index + 1)
       }
     }
+  }
+and evaluate_ap_builtin = (ident: string, args: list(DHExp.t)): result => {
+  switch (ident) {
+  | "int_of_float" =>
```

Comment:
let's decentralize this logic into a lookup table in the Builtins module rather than with a switch here, so new builtins only require updating one file ideally

Reply:
I tried doing this initially, but I got circular module dependency errors (the table in `Builtins` would rely on `Evaluator.result`, and `Evaluator` has to look up in `Builtins`. One solution would be to move `Evaluator.result` into its own module?

Reply:
Yeah moving Evaluator.result into EvaluatorResult.t makes sense.

Reply:
actually calling it EvaluationResult.t is better so that we can reuse it even when we have a compiler

Reply:
Cool, I'll do that